rnd/autogpt_server/autogpt_server/server/routers/analytics.py

```diff
@@ -0,0 +1,138 @@
+"""Analytics API"""
+
+from typing import Annotated, Optional
+
+import fastapi
+import prisma
+import prisma.enums
+import pydantic
+
+import autogpt_server.data.analytics
+from autogpt_server.server.utils import get_user_id
+
+router = fastapi.APIRouter()
+
+
+class UserData(pydantic.BaseModel):
+    user_id: str
+    email: str
+    name: str
+    username: str
+
+
+@router.post(path="/log_new_user")
+async def log_create_user(
+    user_id: Annotated[str, fastapi.Depends(get_user_id)],
+    user_data: Annotated[
+        UserData, fastapi.Body(..., embed=True, description="The user data to log")
+    ],
+):
+    """
+    Log the user ID for analytics purposes.
+    """
+
+    result = await autogpt_server.data.analytics.log_raw_analytics(
+        user_id,
+        prisma.enums.AnalyticsType.CREATE_USER,
+        user_data.model_dump(),
+        "",
+    )
+    return result.id
+
+
+@router.post(path="/log_tutorial_step")
+async def log_tutorial_step(
+    user_id: Annotated[str, fastapi.Depends(get_user_id)],
+    step: Annotated[str, fastapi.Body(..., embed=True)],
+    data: Annotated[
+        Optional[dict],
+        fastapi.Body(..., embed=True, description="Any additional data to log"),
+    ],
+):
+    """
+    Log the tutorial step completed by the user for analytics purposes.
+    """
+    result = await autogpt_server.data.analytics.log_raw_analytics(
+        user_id,
+        prisma.enums.AnalyticsType.TUTORIAL_STEP,
+        data or {},
+        step,
+    )
+    await autogpt_server.data.analytics.log_raw_metric(
+        user_id=user_id,
+        metric_name=prisma.enums.AnalyticsMetric.TUTORIAL_STEP_COMPLETION,
+        aggregation_type=prisma.enums.AggregationType.COUNT,
+        metric_value=1,
+        data_string=step,
+    )
+    return result.id
+
+
+class PageViewData(pydantic.BaseModel):
+    page: str = pydantic.Field(description="The page viewed")
+    data: Optional[dict] = pydantic.Field(
+        default_factory=dict, description="Any additional data to log"
+    )
+
+
+@router.post(path="/log_page_view")
+async def log_page_view(
+    user_id: Annotated[str, fastapi.Depends(get_user_id)],
```

Comment:
Give most all of our pages have an api call, can we not add this logging to these api calls.

Reply:
Yeah sure. The point of this is custom analytics and debugging. The goal is to be able to provide custom data packets to page views, not just a single page view. Sentry is much better for that.